test/parallel/test-assert.js

```diff
@@ -1078,3 +1084,15 @@ assert.throws(
     }
   );
 }
+
+assert.throws(
+  () => assert.throws(
+    () => { throw new TypeError('foobar'); },
+    { name: assert.undefinedProperty }
+  ),
+  {
+    message: `${start}\n${actExp}\n\n` +
+             "+ Comparison {\n+   name: 'TypeError'\n+ }\n" +
+             '- Comparison {}'
```

Comment:
nit: Any reason why this string shouldn't be constructed using just template literals? Seems like that would be more consistent.

```javascript
message: `${start}\n${actExp}\n\n` +
         `+ Comparison {\n+   name: 'TypeError'\n+ }\n` +
         `- Comparison {}`
```

Reply:
Thanks for clarifying! 👍